### PR TITLE
refactor lexer storage and setter methods

### LINKED ISSUE #78: [Maint] Refactor Lexer storage and setter methods
**Is your proposal related to a specific pain point or technical debt? Please describe.**
Right now, our lexer unit tests rely on reading physical files from disk (`data/load_image.pic`, `data/numbers.pic`, etc.). This introduces minor technical debt:

- Tests are harder to read because the test input is separated from the test assertions.
- Tests are fragile because relative file paths can break depending on where the test runner is executed.
- Testing small edge cases requires creating and managing lots of tiny files on disk.

**Describe the proposed changes**
Add a `setSourceString` method to our `Lexer` class that loads source code directly from an in-memory string instead of a file on disk. `setSource` will be refactored to read the file into memory and delegate to `setSourceString`.

Expected interface in `Lexer.h`:

```cpp
class Lexer {
public:
    // Set source directly from memory (great for GoogleTest!)
    void setSourceString(std::string_view source);

    // Reads a file from disk into memory, then calls setSourceString()
    std::expected<void, LexerError> setSource(const std::string& filepath);

private:
    std::string _buffer;  // Single source of truth holding the active source code
    size_t _position{0};
    size_t _line{1};
    size_t _column{1};
};
```

In `Lexer.cpp`:

- `setSourceString(std::string_view source)` assigns `source` to `_buffer` and resets state (`_position = 0`, `_line = 1`, `_column = 1`).
- `setSource(const std::string& filepath)` reads the file into a local string buffer, then calls `setSourceString(...)`.

**Expected impact and benefits**
- **Cleaner Unit Tests:** Source code inputs can be written inline right next to assertions using string literals.
- **Faster Executions & Zero File I/O in UTs:** Unit tests run entirely in memory without hitting disk I/O.
- **Easier Test Maintenance:** Adding new lexer edge cases won't require creating new files in `data/`.

**Steps to implement / Checklist**
- [x] Add `setSourceString(std::string_view source)` to `Lexer.h` and `Lexer.cpp`.
- [x] Refactor `Lexer::setSource` to read the file into a string and call `setSourceString`.
- [x] Update existing tests in `lexer_test.cpp` to use `setSourceString` with inline string literals.
- [x] Remove deprecated `.pic` test files from the `data/` folder if no longer needed.
- [x] Ensure all GoogleTest cases pass.

**Additional context**
Example of how `lexer_test.cpp` will look after this refactor:

```cpp
TEST_F(LexerTest, LoadImageStatement) {
    // Pass the source directly as an inline string!
    _lexer.setSourceString("img = load_image(\"cat.jpg\")");

    auto tokensRes = _lexer.tokenizeAll();
    ASSERT_TRUE(tokensRes.has_value());

    const auto &tokens = tokensRes.value();
    ASSERT_GE(tokens.size(), 7);
    EXPECT_EQ(tokens[0].type(), picceler::Token::Type::IDENTIFIER);
    EXPECT_EQ(tokens[0].value(), "img");
    // ...
}
```

### PR DESCRIPTION
**Pull request by emdevv**
Fixes #78 !

### CODE REVIEW

File: include/lexer.h [hunk 1]
```diff
@@ -114,12 +114,14 @@ class Lexer {
    * @brief Constructs a Lexer
    */
   Lexer();
-
+  ////////
```
Comment (Robertkq) [inline include/lexer.h]:
```suggestion
  /**
  * @brief ....
  * @param source ...
  */
```

Why the `////////` ? Instead lets use `doxygen` style comments, that will be used to generate documentation automatically, see https://robertkq.github.io/picceler/

File: include/lexer.h [hunk 2]
```diff
@@ -145,6 +147,9 @@ class Lexer {
   Result<std::vector<Token>> tokenizeAll();
 
 private:
+  /////////
```
Comment (Robertkq) [inline include/lexer.h]: same here, why `/////////` ? use doxygen instead if you want to document it

File: include/parser.h [hunk 3]
```diff
@@ -23,6 +23,12 @@ class Parser {
    * @brief Sets the source file for the parser.
    * @param source The source file to read from.
    */
+  void setSourceString(std::string_view source);
```
Comment (Robertkq) [inline include/parser.h:26]: yea i didnt think of this but i think its good to have a `setSourceString` in the parser as well, so that we can specify the same way for parser tests

File: tests/unit/src/lexer_tests.cpp [hunk 4]
```diff
@@ -10,33 +10,24 @@ class LexerTest : public ::testing::Test {
...
 
   ASSERT_EQ(tokens.size(), 1);
   EXPECT_EQ(tokens[0].type(), picceler::Token::Type::EOF_TOKEN);
 }
 
 TEST_F(LexerTest, LoadImageStatement) {
-  auto res = _lexer.setSource("data/load_image.pic");
-  if (!res)
-    FAIL() << res.error().message();
+  _lexer.setSourceString("img = load_image(\"cat.jpg\")");
```
Comment (Robertkq) [inline tests/unit/src/lexer_tests.cpp]:
```suggestion
  _lexer.setSourceString(R"(img = load_image("cat.jpg"))");
```
`R(<content>)` prefix for a string sets it as raw string literal, meaning it will be printed just as written, no special character treated. You already used this for the empty test case above.. so we can just use it here too. Didn't know this existed before tbh

File: tests/unit/src/parser_tests.cpp [hunk 5]
```diff
@@ -51,12 +49,8 @@ TEST_F(ParserTest, EmptyInput) {
 }
 
 TEST_F(ParserTest, LoadImageStatement) {
-  auto result = _parser.setSource("data/load_image.pic");
-  if (!result)
-    FAIL() << result.error().message();
+  _parser.setSourceString("img = load_image(\"cat.jpg\")");
```
Comment (Robertkq) [inline tests/unit/src/parser_tests.cpp]: Here we can use that `R"(<content>)" again

**Review by Robertkq — CHANGES_REQUESTED** [review]

File: include/parser.h (hunk 3 above)
Comment (emdevv) [reply include/parser.h:26]: solved

File: include/lexer.h (hunk 2 above)
Comment (emdevv) [reply include/lexer.h]: solved

File: include/lexer.h (hunk 1 above)
Comment (emdevv) [reply include/lexer.h]: solved

**Review by Robertkq — APPROVED** [review]
good work! thanks for doing this
